crates/red_knot/src/types.rs

```diff
@@ -484,7 +511,7 @@ mod tests {
 
     #[test]
     fn add_union() {
-        let mut store = TypeStore::default();
+        let store = TypeStore::default();
         let files = Files::default();
         let file_id = files.intern(Path::new("/foo"));
         let c1 = store.add_class(file_id, "C1");
```

Comment:
Nit: Adding some whitespace to these tests could help readability. I'm not a purist that a unit test always needs to be structured in Arrange, Act, and assert, and that each test is only allowed to have one of each section, but I do find that grouping the individual sections helps improve readability (or parseability):

```rust
        let store = TypeStore::default();
        let files = Files::default();
        let file_id = files.intern(Path::new("/foo"));
        
        let c1 = store.add_class(file_id, "C1");
        let c2 = store.add_class(file_id, "C2");
        let elems = vec![Type::Instance(c1), Type::Instance(c2)];
        let id = store.add_union(file_id, &elems);
        
        assert_eq!(
            store.get_union(id).elements,
            elems.into_iter().collect::<FxIndexSet<_>>()
        );
        
        let union = Type::Union(id);
        assert_eq!(format!("{}", union.display(&store)), "(C1 | C2)");
```

Reply:
Good call-out, I usually do try to do that, but for some reason didn't here.